### ISSUE TITLE
How to run WebDriver::attachFile in Docker?

### ISSUE THREAD

**Issue by Luc45** [Codeception/Codeception#5809]
#### What are you trying to achieve?

I'm trying to upload a file using \Codeception\Module\WebDriver::attachFile while in the context of Docker containers.

#### What do you get instead?

Selenium can't find my file.

```bash
[Facebook\WebDriver\Exception\InvalidArgumentException] invalid argument: File not found : /application/www/dev/tests/tests/_data/800x800.png
  (Session info: headless chrome=75.0.3770.80)
```

> Provide test source code if related

```php
$I->attachFile( '.dz-hidden-input', '800x800.png' );
```

Going into attachFile, I see this:

```php
    public function attachFile($field, $filename)
    {
        $el = $this->findField($field);
        // in order to be compatible on different OS
        $filePath = codecept_data_dir() . $filename;
        if (!file_exists($filePath)) {
            throw new \InvalidArgumentException("File does not exist: $filePath");
        }
        if (!is_readable($filePath)) {
            throw new \InvalidArgumentException("File is not readable: $filePath");
        }
        // in order for remote upload to be enabled
        $el->setFileDetector(new LocalFileDetector());

        // skip file detector for phantomjs
        if ($this->isPhantom()) {
            $el->setFileDetector(new UselessFileDetector());
        }
        $el->sendKeys(realpath($filePath));
    }
```

It does get past `(!file_exists($filePath)`, because when in the context that the PHP interpreter on Codeception runs, the file exists and it can see it. However, when it delegates to the WebDriver the action of attaching the file, on line `$el->sendKeys(realpath($filePath));`, then Selenium ends up not being able to find it because it is a different container.

Due to my Docker Composer configuration, I can't share volumes between Selenium and the PHP container.

Is there a way to get around that, maybe by attaching a file from a remote source or something? I would think this should be a fairly common problem since everyone is moving towards Docker development.

### Details

* Codeception version: 3.1.2
* PHP Version: 7.2
* Operating System: Ubuntu 18.04
* Installation type: Composer

**Comment by nicholascus** [comment]
That sounds like Selenium question rather than codeception one. I know it is addressed in java webdriver client library, but not sure about the Facebook php webdriver client library which is used by Codeception.
https://blogs.perficient.com/2018/08/07/upload-a-file-to-a-random-vm-on-selenium-grid/

**Comment by Naktibalda** [comment]
The code in that article looks like the code of attachFile, there is no difference in it.
Maybe LocalFileDetector does something more in Java than in PHP.

**Comment by Naktibalda** [comment]
I looked at this issue again.
I confirmed that `attachFile` really uploads zipped file to Selenium.

The file must be accessible to Codeception, php-webdriver library uploads files to Selenium

The code of [sendKeys](https://github.com/php-webdriver/php-webdriver/blob/1.8.1/lib/Remote/RemoteWebElement.php#L326-L368) and [upload](https://github.com/php-webdriver/php-webdriver/blob/1.8.1/lib/Remote/RemoteWebElement.php#L495-L511) functions.

It looks like your browser received the path which was valid in Codeception container, that means that $local_files was `null` in `sendKeys` because either LocalFileDetector couldn't see that file using `is_file` function or a UselessFileDetector was used instead.
UselessFileDetector is only used if `browser` setting in configuration starts with `phantom`.

Is there any chance that `/application/www/dev/tests/tests/_data/800x800.png` exists and is readable but it isn't a file?
